src/document/Document.js

```diff
@@ -675,10 +675,12 @@ define(function (require, exports, module) {
     
     /**
      * Updates the language to match the current mapping given by LanguageManager
+     * 
+     * @param {!string} rawText  Text content of the file.
      */
-    Document.prototype._updateLanguage = function () {
+    Document.prototype._updateLanguage = function (rawText) {
```

Comment:
I believe `this.getText()` won't exist cause the functions are called by `Document` contructor.  Therefore,  using `rawText` is probably the best way to pass the contents to `LanguageManager`.  So far, the shebang detection happens only when user opens a file, I am not sure when it triggers during editing.  Could you or anybody help me for that?